pkg/websocket/websocket.go

```diff
@@ -88,9 +88,9 @@ func (ws *websocketClient) ConnectContext(ctx context.Context, addr string, head
 	logrus.Infof("websocket: connected to %s", addr)
 	ws.wasConnected()
 	ws.conn = c
-	ws.wg.Add(2)
-	go ws.readPump()
-	go ws.writePump(ctx)
+	ws.readPump()
+	ws.writePump(ctx) <- struct{}{}
```

Comment:
why the ready stuff here? select case on ready does nothing? I might be missing something....

Reply:
It was because of a race condition. The writePump did not have time to start before the OnConnect event was called and tried to send

Reply:
ah okey nice catch